src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.ILLink.targets

```diff
@@ -0,0 +1,110 @@
+<!--
+***********************************************************************************************
+Microsoft.NET.ILLink.targets
+
+WARNING:  DO NOT MODIFY this file unless you are knowledgeable about MSBuild and have
+          created a backup copy.  Incorrect changes to this file will make it
+          impossible to load or build your projects from the command-line or the IDE.
+
+Copyright (c) .NET Foundation. All rights reserved.
+***********************************************************************************************
+-->
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+
+  <Import Project="Sdk.props" Sdk="ILLink.Tasks" />
+
+  <PropertyGroup>
+    <ILTransformDependsOn>$(ILTransformDependsOn);ILLink</ILTransformDependsOn>
+    <IntermediateLinkDir Condition=" '$(IntermediateLinkDir)' == '' ">$(IntermediateOutputPath)linked</IntermediateLinkDir>
+    <!-- Used to enable incremental build for the linker target. -->
+    <_LinkSemaphore>$(IntermediateOutputPath)Link.semaphore</_LinkSemaphore>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <LinkerRootDescriptors Include="$(LinkerRootDescriptors)" />
+  </ItemGroup>
+
+  <!--
+    ============================================================
+                     ILLink
+
+    Replace the files to be published with versions that have been
+    passed through the linker. Also prevent removed files from being
+    included in the generated deps.json.
+    ============================================================
+    -->
+  <Target Name="ILLink"
+          Condition=" '$(LinkDuringPublish)' == 'true' "
+          DependsOnTargets="RunILLink">
+
+    <!-- For now, use ResolvedFileToPublish as input/output. This
```

Comment:
The linker works with `ResolvedFileToPublish` as input, and I think that's the way it should be for now.

But I'm curious why crossgen works differently. Crossgen targets run after `_ComputeResolvedFilesToPublishTypes` target, and processes/updates the files bifurcated into `_ResolvedFileToPublishPreserveNewest`  and `_ResolvedFileToPublishAlways` items. However, the processing /input-outputs for the two lists there are similar. So, why doesn't crossgen similarly process `ResolvedFileToPublish` ?

Reply:
Good question - @fadimounir can you answer this?

Reply:
When I started implementing this, i was under the impression that I would need to add some logic to the crossgen task to distinguish between the 2 lists, and only compile things in the PreserveNewest if they were out of date.
Things have evolved since then, and thinking about it now, I believe we should be able to just make crossgen work with the 'ResolvedFileToPublish', before the split that '_ComputeResolvedFilesToPublishTypes' does.
I'll make that change. For the linker, i think this looks correct.
The linker does one operation with multiple input files. Crossgen does multiple operations, each with its own input file, so splitting the 'ResolvedFilesToPublish' by type doesn't make sense here i believe